models/HubSpotHttpError.ts

```diff
@@ -116,7 +121,7 @@ export class HubSpotHttpError<T = any> extends Error {
     generatedContext.request = cause.config?.url;
 
     // Allow the provided context to override the generated context
-    this.context = { ...generatedContext, ...this.context };
+    this.derivedContext = { ...this.derivedContext, ...generatedContext };
```

Comment:
Splitting this into `derivedContext` to prevent collisions with the user provided context and it doesn't unintentionally slip unwanted details into the `message`.